cobigen-plugins/cobigen-templateengines/cobigen-tempeng-freemarker/src/main/java/com/devonfw/cobigen/tempeng/freemarker/FreeMarkerTemplateEngine.java

```diff
@@ -73,8 +73,14 @@ public void process(TextTemplate template, Map<String, Object> model, Writer out
         env.setLogTemplateExceptions(false); // no duplicate logging
         env.process();
       } catch (TemplateException e) {
+        String detailedCause = "";
+        if (e.getCause().getClass().getCanonicalName().contains("java.lang")) {
```

Comment:
Just getting the first cause is maybe not what you want. Most probably you want to look at the root cause wouldn't you?

Reply:
True! Next commit extracts the rootcause before building error message.